webapp/src/ts/services/telemetry.service.ts

```diff
@@ -263,29 +267,35 @@ export class TelemetryService {
     return this.ngZone.runOutsideAngular(() => this._record(key, value));
   }
 
-  private _record(key, value?) {
+  private async _record(key, value?) {
     if (value === undefined) {
       value = 1;
     }
 
-    let db;
-    this.queue = this.queue
-      .then(() => db = this.getDb())
-      .then(() => this.submitIfNeeded(db))
-      .then(() => db = this.getDb())  // db is fetched again in case submitIfNeeded dropped the old reference
-      .then(() => this.storeIt(db, key, value))
-      .catch(err => console.error('Error in telemetry service', err))
-      .finally(() => {
-        if (!db || db._destroyed || db._closed) {
-          return;
-        }
-        try {
-          db.close();
-        } catch (err) {
-          console.error('Error closing telemetry DB', err);
-        }
-      });
+    let currentDB;
+    try {
+      const today = this.getToday();
+      const telemetryDBs = await this.getTelemetryDBs();
+      await this.submitIfNeeded(today, telemetryDBs);
+      currentDB = await this.getCurrentTelemetryDB(today);
+      await this.storeIt(currentDB, key, value);
+    } catch (error) {
+      console.error('Error in telemetry service', error);
+    }
 
-    return this.queue;
+    if (!currentDB || currentDB?._destroyed || currentDB?._closed) {
+      return;
+    }
+
+    try {
+      currentDB.close();
+    } catch (error) {
+      console.error('Error closing telemetry DB', error);
+    }
```

Comment:
Minor, but I always try to avoid `let`.  Would something like this be acceptable?

```suggestion
    try {
      const today = this.getToday();
      const telemetryDBs = await this.getTelemetryDBs();
      await this.submitIfNeeded(today, telemetryDBs);
      const currentDB = await this.getCurrentTelemetryDB(today);
      try {
        await this.storeIt(currentDB, key, value);
      } finally {
        try {
          currentDB.close();
        } catch (error) {
          console.error('Error closing telemetry DB', error);
        }
      }
    } catch (error) {
      console.error('Error in telemetry service', error);
    }
```

Reply:
Yeah, I was back and forth on this, I tried to keep it flat and seemed simpler. I think a bit more how to make it prettier

Reply:
I think I got the best of both worlds. 😬